tests/PackageUrl.Tests.cs

```diff
@@ -80,5 +81,39 @@ public void TestConstructorParameters(PurlTestData data)
                 Assert.NotNull(purl.Qualifiers);
             }
         }
+        
+        [Theory]
+        [InlineData("{\"Scheme\":\"pkg\",\"Type\":\"npm\",\"Namespace\":null,\"Name\":\"foo\"," +
+                    "\"Version\":\"1.2.3\",\"Qualifiers\":null,\"Subpath\":null}",
+            "pkg:npm/foo@1.2.3")]
+        public void TestJsonSerialization(string serializedPurl, string packageUrlStr)
```

Comment:
There should be negative test cases added here as well to verify the negative cases where exceptions should be thrown.